fastlane/lib/fastlane/actions/upload_app_data_usage_to_app_store.rb

```diff
@@ -0,0 +1,316 @@
+          # Ask if using category
+          next unless UI.confirm("Collect data for #{category.id}?")
+
+          purpose_names = purposes.map(&:id).join(', ')
+          UI.message("How will this data be used? You'll be offered with #{purpose_names}")
+
+          # Ask purposes
+          selected_purposes = []
+          purposes.each do |purpose|
+            selected_purposes << purpose if UI.confirm("Used for #{purpose.id}?")
+          end
+
+          # Ask protections
+          is_linked_to_user = UI.confirm("Is #{category.id} linked to the user?")
+          is_used_for_tracking = UI.confirm("Is #{category.id} used for tracking purposes?")
+
+          # Map answers to values for API requests
+          protection_id = is_linked_to_user ? "DATA_LINKED_TO_YOU" : "DATA_NOT_LINKED_TO_YOU"
+          tracking_id = is_used_for_tracking ? "DATA_USED_TO_TRACK_YOU" : nil
+
+          json << {
+            "category" => category.id,
+            "purposes" => selected_purposes.map(&:id),
+            "data_protections" => [
+              protection_id, tracking_id
+            ].compact
+          }
+        end
+
+        return json
+      end
+
+      def self.upload_app_data_usages(params, app, usages_config)
+        # Delete all existing usages for new ones
+        all_usages = Spaceship::ConnectAPI::AppDataUsage.all(app_id: app.id, includes: "category,grouping,purpose,dataProtection", limit: 500)
+        all_usages.each(&:delete!)
+
+        usages_config.each do |usage_config|
+          category = usage_config["category"]
+          purposes = usage_config["purposes"] || []
+          data_protections = usage_config["data_protections"] || []
+
+          # There will not be any purposes if "not collecting data"
+          # However, an AppDataUsage still needs to be created for not collecting data
+          # Creating an array with nil so that purposes can be iterated over and
+          # that AppDataUsage can be created
+          purposes = [nil] if purposes.empty?
+
+          purposes.each do |purpose|
+            data_protections.each do |data_protection|
+              UI.message("Setting #{category} and #{purpose} to #{data_protection}")
+              Spaceship::ConnectAPI::AppDataUsage.create(
+                app_id: app.id,
+                app_data_usage_category_id: category,
+                app_data_usage_protection_id: data_protection,
+                app_data_usage_purpose_id: purpose
+              )
+            end
+          end
+        end
+
+        # Publish
+        publish_state = Spaceship::ConnectAPI::AppDataUsagesPublishState.get(app_id: app.id)
+        if publish_state.published
+          UI.important("App data usage is already published")
+        else
+          UI.important("App data usage not published! Going to publish...")
+          publish_state.publish!
+          UI.important("App data usage is now published")
+        end
+      end
+
+      def self.ask_interactive_questions(app)
```

Comment:
Looks like this method may have been replaced by `ask_interactive_questions_for_json`

Reply:
lol yup, forgot to delete 😛